[bug] FreeBSD compilation fails on scrypt.h

# Bug Report

**Current behavior:**

Compiling Dogecoin Core (without wallet) on FreeBSD 11.4 fails with the following error:

```
  CXX      primitives/libdogecoin_common_a-pureheader.o 

In file included from primitives/pureheader.cpp:9:                              
./crypto/scrypt.h:30:24: error: redefinition of 'le32dec'                       
static inline uint32_t le32dec(const void *pp)                                  
                       ^                                                        
/usr/include/sys/endian.h:132:1: note: previous definition is here              
le32dec(const void *pp)                                                         
^                                                                               
In file included from primitives/pureheader.cpp:9:                              
./crypto/scrypt.h:37:20: error: redefinition of 'le32enc'                       
static inline void le32enc(void *pp, uint32_t x)                                
                   ^                                                            
/usr/include/sys/endian.h:186:1: note: previous definition is here              
le32enc(void *pp, uint32_t u)                                                   
^                                                                               
2 errors generated.                   
```

**Steps to reproduce:**

On a clean FreeBSD 11.4 install, do:

```bash
pkg install autoconf automake libtool pkgconf
pkg install boost-libs openssl libevent
pkg install gmake
pkg install git
git clone https://github.com/dogecoin/dogecoin.git
cd dogecoin
./autogen.sh 
./configure --disable-wallet
gmake
```

**Dogecoin Core version**: 1.14.4

**Machine specs**
- OS: FreeBSD 11.4
- CPU: QEMU Virtual CPU version 2.5+ (amd64)
- RAM: 2048MB

> # Bug Report
> 
> 
> 
> **Current behavior:**
> 
> 
> 
> Compiling Dogecoin Core (without wallet) on FreeBSD 11.4 fails with the following error:
> 
> 
> 
> ```
> 
>   CXX      primitives/libdogecoin_common_a-pureheader.o 
> 
> 
> 
> In file included from primitives/pureheader.cpp:9:                              
> 
> ./crypto/scrypt.h:30:24: error: redefinition of 'le32dec'                       
> 
> static inline uint32_t le32dec(const void *pp)                                  
> 
>                        ^                                                        
> 
> /usr/include/sys/endian.h:132:1: note: previous definition is here              
> 
> le32dec(const void *pp)                                                         
> 
> ^                                                                               
> 
> In file included from primitives/pureheader.cpp:9:                              
> 
> ./crypto/scrypt.h:37:20: error: redefinition of 'le32enc'                       
> 
> static inline void le32enc(void *pp, uint32_t x)                                
> 
>                    ^                                                            
> 
> /usr/include/sys/endian.h:186:1: note: previous definition is here              
> 
> le32enc(void *pp, uint32_t u)                                                   
> 
> ^                                                                               
> 
> 2 errors generated.                   
> 
> ```
> 
> 
> 
> **Steps to reproduce:**
> 
> 
> 
> On a clean FreeBSD 11.4 install, do:
> 
> 
> 
> ```bash
> 
> pkg install autoconf automake libtool pkgconf
> 
> pkg install boost-libs openssl libevent
> 
> pkg install gmake
> 
> pkg install git
> 
> git clone https://github.com/dogecoin/dogecoin.git
> 
> cd dogecoin
> 
> ./autogen.sh 
> 
> ./configure --disable-wallet
> 
> gmake
> 
> ```
> 
> 
> 
> **Dogecoin Core version**: 1.14.4
> 
> 
> 
> **Machine specs**
> 
> - OS: FreeBSD 11.4
> 
> - CPU: QEMU Virtual CPU version 2.5+ (amd64)
> 
> - RAM: 2048MB
> 
>

--- a/src/crypto/scrypt.cpp
+++ b/src/crypto/scrypt.cpp
@@ -44,6 +44,7 @@
 #endif
 #endif
 
+#ifndef __FreeBSD__
 static inline uint32_t be32dec(const void *pp)
 {
 	const uint8_t *p = (uint8_t const *)pp;
@@ -60,6 +61,7 @@ static inline void be32enc(void *pp, uint32_t x)
 	p[0] = (x >> 24) & 0xff;
 }
 
+#endif
 /**
  * PBKDF2_SHA256(passwd, passwdlen, salt, saltlen, c, buf, dkLen):
  * Compute PBKDF2(passwd, salt, c, dkLen) using HMAC-SHA256 as the PRF, and

--- a/src/crypto/scrypt.h
+++ b/src/crypto/scrypt.h
@@ -27,6 +27,7 @@ void
 PBKDF2_SHA256(const uint8_t *passwd, size_t passwdlen, const uint8_t *salt,
     size_t saltlen, uint64_t c, uint8_t *buf, size_t dkLen);
 
+#ifndef __FreeBSD__
 static inline uint32_t le32dec(const void *pp)
 {
         const uint8_t *p = (uint8_t const *)pp;
@@ -42,4 +43,5 @@ static inline void le32enc(void *pp, uint32_t x)
         p[2] = (x >> 16) & 0xff;
         p[3] = (x >> 24) & 0xff;
 }
+#endif
 #endif

